CI Failure (ubuntu)

`Error: Failed with code Some(1): ./ninja extract:protoc css:_root-vars ts:reviewer:reviewer.js ts:reviewer:reviewer.css ts:reviewer:reviewer_extras_bundle.js ts:reviewer:reviewer_extras.css qt:aqt:data:web:pages`


* Last success: 2024-04-29T01:46:25+01:00 - https://github.com/ankidroid/Anki-Android-Backend/actions/runs/8871673583/job/24354979002
* First failure: 2024-04-30T01:44:57+01:00 https://github.com/ankidroid/Anki-Android-Backend/actions/runs/8887480874/job/24402827438

**good**
```
2024-04-29T00:49:16.7238937Z ##[group]Run cargo run -p build_rust
2024-04-29T00:49:16.7239368Z [36;1mcargo run -p build_rust[0m
2024-04-29T00:49:16.7289573Z shell: /usr/bin/bash -e {0}
2024-04-29T00:49:16.7289911Z env:
2024-04-29T00:49:16.7290151Z   DEBUG: 1
2024-04-29T00:49:16.7290539Z   ANDROID_NDK_HOME: /usr/local/lib/android/sdk/ndk/26.3.11579264
2024-04-29T00:49:16.7291302Z   ANDROID_NDK_ROOT: /usr/local/lib/android/sdk/ndk/26.3.11579264
2024-04-29T00:49:16.7291905Z   JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/21.0.2-13/x64
2024-04-29T00:49:16.7292585Z   JAVA_HOME_21_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/21.0.2-13/x64
2024-04-29T00:49:16.7293147Z   GRADLE_ACTION_ID: gradle/actions/setup-gradle
2024-04-29T00:49:16.7293586Z   GRADLE_BUILD_ACTION_SETUP_COMPLETED: true
2024-04-29T00:49:16.7293990Z   GRADLE_BUILD_ACTION_CACHE_RESTORED: true
2024-04-29T00:49:16.7294584Z   DEVELOCITY_INJECTION_INIT_SCRIPT_NAME: gradle-actions.inject-develocity.init.gradle
2024-04-29T00:49:16.7295290Z   DEVELOCITY_AUTO_INJECTION_CUSTOM_VALUE: gradle-actions
2024-04-29T00:49:16.7295755Z   GITHUB_DEPENDENCY_GRAPH_ENABLED: false
2024-04-29T00:49:16.7296121Z ##[endgroup]
2024-04-29T00:49:16.8932910Z    Compiling anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
2024-04-29T00:49:16.8934953Z    Compiling anki_process v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/process)
2024-04-29T00:49:16.9575255Z    Compiling build_rust v0.1.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/build_rust)
2024-04-29T00:49:17.2798352Z     Finished dev [unoptimized + debuginfo] target(s) in 0.51s
2024-04-29T00:49:17.2840274Z      Running `target/debug/build_rust`
2024-04-29T00:49:17.2884390Z *** Building desktop web components
2024-04-29T00:49:17.5103476Z [1m[32m   Compiling[0m runner v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/build/runner)
2024-04-29T00:49:17.5105983Z [1m[32m   Compiling[0m anki_process v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/process)
2024-04-29T00:49:17.5107952Z [1m[32m   Compiling[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
2024-04-29T00:49:21.1401393Z [1m[32m    Finished[0m release [optimized] target(s) in 3.81s
2024-04-29T00:49:21.3024538Z [1m[32m   Compiling[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
2024-04-29T00:49:21.3465267Z [1m[32m   Compiling[0m ninja_gen v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/build/ninja_gen)
2024-04-29T00:49:21.5517166Z [1m[32m   Compiling[0m configure v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/build/configure)
2024-04-29T00:49:22.1599019Z [1m[32m    Finished[0m dev [unoptimized + debuginfo] target(s) in 0.97s
2024-04-29T00:49:22.1656173Z [1m[32m     Running[0m `out/rust/debug/configure`
2024-04-29T00:49:22.2441737Z build:configure_bin
2024-04-29T00:49:22.4218422Z [1m[32m    Finished[0m dev [unoptimized + debuginfo] target(s) in 0.13s
2024-04-29T00:49:22.4219452Z build:configure
2024-04-29T00:49:22.4932216Z download:protoc
2024-04-29T00:49:22.4933217Z download:node
2024-04-29T00:49:22.4939533Z ftl:repo:core
2024-04-29T00:49:22.5010171Z extract:protoc
2024-04-29T00:49:22.5283361Z extract:node
2024-04-29T00:49:22.5291951Z rslib:proto
2024-04-29T00:49:24.6573450Z yarn
2024-04-29T00:49:24.7701876Z node_modules
2024-04-29T00:49:24.9033090Z rslib:proto
2024-04-29T00:49:24.9034566Z [1m[32m   Compiling[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
2024-04-29T00:49:24.9036558Z [1m[32m   Compiling[0m anki_proto_gen v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/proto_gen)
2024-04-29T00:49:24.9038972Z [1m[32m   Compiling[0m anki_proto v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/proto)
2024-04-29T00:49:24.9040796Z [1m[32m    Finished[0m dev [unoptimized + debuginfo] target(s) in 2.32s
2024-04-29T00:49:25.7729776Z ftl:repo:core
2024-04-29T00:49:25.7731981Z Submodule 'ftl/core-repo' (https://github.com/ankitects/anki-core-i18n.git) registered for path 'ftl/core-repo'
2024-04-29T00:49:25.7734463Z Cloning into '/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/ftl/core-repo'...
2024-04-29T00:49:25.7736312Z From https://github.com/ankitects/anki-core-i18n
2024-04-29T00:49:25.7737936Z  * branch              e3af3c983241448a239871ca573c9dd2fa5e8619 -> FETCH_HEAD
2024-04-29T00:49:25.7739795Z Submodule path 'ftl/core-repo': checked out 'e3af3c983241448a239871ca573c9dd2fa5e8619'
2024-04-29T00:49:25.7741449Z ftl:repo:qt
2024-04-29T00:49:25.9332507Z node_modules
2024-04-29T00:49:25.9333373Z [2K[1G[1myarn install v1.22.22[22m
2024-04-29T00:49:25.9334143Z [2K[1G[2m[1/4][22m Resolving packages...
2024-04-29T00:49:25.9334926Z [2K[1G[32msuccess[39m Already up-to-date.
2024-04-29T00:49:25.9335587Z [2K[1GDone in 0.36s.
2024-04-29T00:49:25.9339367Z ts:lib:proto
2024-04-29T00:49:25.9339912Z ts:lib:proto
2024-04-29T00:49:25.9340571Z css:_root-vars
2024-04-29T00:49:26.7015197Z ts:reviewer:reviewer_extras.css
2024-04-29T00:49:27.0289477Z ts:reviewer:reviewer.css
2024-04-29T00:49:27.7258427Z ftl:repo:qt
2024-04-29T00:49:27.7259941Z Submodule 'ftl/qt-repo' (https://github.com/ankitects/anki-desktop-ftl.git) registered for path 'ftl/qt-repo'
2024-04-29T00:49:27.7261779Z Cloning into '/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/ftl/qt-repo'...
2024-04-29T00:49:27.7263121Z From https://github.com/ankitects/anki-desktop-ftl
2024-04-29T00:49:27.7264330Z  * branch            45155310c3302cbbbe645dec52ca196894422463 -> FETCH_HEAD
2024-04-29T00:49:27.7265667Z Submodule path 'ftl/qt-repo': checked out '45155310c3302cbbbe645dec52ca196894422463'
2024-04-29T00:49:27.7266603Z rslib:i18n
2024-04-29T00:49:29.5202965Z [1m[32m   Compiling[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
2024-04-29T00:49:29.5204200Z [1m[32m   Compiling[0m anki_i18n v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/i18n)
2024-04-29T00:49:29.5205115Z [1m[32m    Finished[0m dev [unoptimized + debuginfo] target(s) in 1.74s
2024-04-29T00:49:29.5209092Z ts:import-csv
2024-04-29T00:49:29.5209658Z ts:image-occlusion
2024-04-29T00:49:29.5210146Z ts:graphs
2024-04-29T00:49:29.5211458Z ts:import-page
2024-04-29T00:49:35.0102943Z ts:congrats
2024-04-29T00:49:35.4381034Z ts:editable
2024-04-29T00:49:36.9213228Z ts:reviewer:reviewer_extras_bundle.js
2024-04-29T00:49:37.4991293Z ts:change-notetype
2024-04-29T00:49:38.0490703Z ts:card-info
2024-04-29T00:49:38.9871136Z ts:deck-options
2024-04-29T00:49:39.2300996Z ts:reviewer:reviewer.js
2024-04-29T00:49:39.7991112Z ts:import-anki-package
2024-04-29T00:49:46.1913594Z qt:aqt:data:web:pages
2024-04-29T00:49:46.2112994Z n2: ran 30 tasks, now up to date
2024-04-29T00:49:46.2118154Z [0m[1m[32m
2024-04-29T00:49:46.2118809Z Build succeeded in 23.98s.
2024-04-29T00:49:46.2687825Z [0m*** Building Android JNI library + backend interface
...
```

**bad**
```

2024-04-30T00:47:18.8748203Z ##[group]Run cargo run -p build_rust
2024-04-30T00:47:18.8748621Z [36;1mcargo run -p build_rust[0m
2024-04-30T00:47:18.8799702Z shell: /usr/bin/bash -e {0}
2024-04-30T00:47:18.8800031Z env:
2024-04-30T00:47:18.8800501Z   DEBUG: 1
2024-04-30T00:47:18.8800871Z   ANDROID_NDK_HOME: /usr/local/lib/android/sdk/ndk/26.3.11579264
2024-04-30T00:47:18.8801415Z   ANDROID_NDK_ROOT: /usr/local/lib/android/sdk/ndk/26.3.11579264
2024-04-30T00:47:18.8802023Z   JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/21.0.2-13/x64
2024-04-30T00:47:18.8802687Z   JAVA_HOME_21_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/21.0.2-13/x64
2024-04-30T00:47:18.8803254Z   GRADLE_ACTION_ID: gradle/actions/setup-gradle
2024-04-30T00:47:18.8803688Z   GRADLE_BUILD_ACTION_SETUP_COMPLETED: true
2024-04-30T00:47:18.8804095Z   GRADLE_BUILD_ACTION_CACHE_RESTORED: true
2024-04-30T00:47:18.8804667Z   DEVELOCITY_INJECTION_INIT_SCRIPT_NAME: gradle-actions.inject-develocity.init.gradle
2024-04-30T00:47:18.8805377Z   DEVELOCITY_AUTO_INJECTION_CUSTOM_VALUE: gradle-actions
2024-04-30T00:47:18.8805830Z   GITHUB_DEPENDENCY_GRAPH_ENABLED: false
2024-04-30T00:47:18.8806188Z ##[endgroup]
2024-04-30T00:47:19.0437931Z    Compiling anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
2024-04-30T00:47:19.0439916Z    Compiling anki_process v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/process)
2024-04-30T00:47:19.1046756Z    Compiling build_rust v0.1.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/build_rust)
2024-04-30T00:47:19.4225109Z     Finished dev [unoptimized + debuginfo] target(s) in 0.50s
2024-04-30T00:47:19.4270234Z      Running `target/debug/build_rust`
2024-04-30T00:47:19.4316260Z *** Building desktop web components
2024-04-30T00:47:19.6519837Z [1m[32m   Compiling[0m runner v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/build/runner)
2024-04-30T00:47:19.6522816Z [1m[32m   Compiling[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
2024-04-30T00:47:19.6525241Z [1m[32m   Compiling[0m anki_process v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/process)
2024-04-30T00:47:23.1136402Z [1m[32m    Finished[0m release [optimized] target(s) in 3.64s
2024-04-30T00:47:23.2741295Z [1m[32m   Compiling[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
2024-04-30T00:47:23.3177155Z [1m[32m   Compiling[0m ninja_gen v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/build/ninja_gen)
2024-04-30T00:47:23.5203477Z [1m[32m   Compiling[0m configure v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/build/configure)
2024-04-30T00:47:24.1162675Z [1m[32m    Finished[0m dev [unoptimized + debuginfo] target(s) in 0.95s
2024-04-30T00:47:24.1218761Z [1m[32m     Running[0m `out/rust/debug/configure`
2024-04-30T00:47:24.1972481Z build:configure_bin
2024-04-30T00:47:24.3687853Z [1m[32m    Finished[0m dev [unoptimized + debuginfo] target(s) in 0.13s
2024-04-30T00:47:24.3688806Z build:configure
2024-04-30T00:47:24.6455112Z [1m[32m    Finished[0m dev [unoptimized + debuginfo] target(s) in 0.11s
2024-04-30T00:47:24.6475951Z [1m[32m     Running[0m `out/rust/debug/configure`
2024-04-30T00:47:24.7213419Z build:configure
2024-04-30T00:47:24.7915036Z download:protoc
2024-04-30T00:47:24.7916263Z download:node
2024-04-30T00:47:24.7918484Z ftl:repo:core
2024-04-30T00:47:24.7991265Z extract:protoc
2024-04-30T00:47:24.8290895Z extract:node
2024-04-30T00:47:24.8298110Z rslib:proto
2024-04-30T00:47:26.9522700Z yarn
2024-04-30T00:47:27.0319674Z node_modules
2024-04-30T00:47:27.0559978Z rslib:proto
2024-04-30T00:47:27.0561244Z [1m[32m   Compiling[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
2024-04-30T00:47:27.0562879Z [1m[32m   Compiling[0m anki_proto_gen v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/proto_gen)
2024-04-30T00:47:27.0564155Z [1m[32m   Compiling[0m anki_proto v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/proto)
2024-04-30T00:47:27.0565052Z [1m[32m    Finished[0m dev [unoptimized + debuginfo] target(s) in 2.18s
2024-04-30T00:47:28.0686848Z ftl:repo:core
2024-04-30T00:47:28.0688546Z Submodule 'ftl/core-repo' (https://github.com/ankitects/anki-core-i18n.git) registered for path 'ftl/core-repo'
2024-04-30T00:47:28.0690419Z Cloning into '/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/ftl/core-repo'...
2024-04-30T00:47:28.0691748Z From https://github.com/ankitects/anki-core-i18n
2024-04-30T00:47:28.0692945Z  * branch              e3af3c983241448a239871ca573c9dd2fa5e8619 -> FETCH_HEAD
2024-04-30T00:47:28.0694282Z Submodule path 'ftl/core-repo': checked out 'e3af3c983241448a239871ca573c9dd2fa5e8619'
2024-04-30T00:47:28.0695255Z ftl:repo:qt
2024-04-30T00:47:28.1699397Z [0m[1m[31m
2024-04-30T00:47:28.1699811Z Build failed.
2024-04-30T00:47:28.1703344Z Error: Failed with code Some(1): ./ninja extract:protoc css:_root-vars ts:reviewer:reviewer.js ts:reviewer:reviewer.css ts:reviewer:reviewer_extras_bundle.js ts:reviewer:reviewer_extras.css qt:aqt:data:web:pages
2024-04-30T00:47:28.1707683Z [0m
2024-04-30T00:47:28.1721715Z ##[error]Process completed with exit code 1.
2024-04-30T00:47:28.1846875Z Post job cleanup.
2024-04-30T00:47:28.4148933Z In post-action step
2024-04-30T00:47:28.4156449Z Stopping all Gradle daemons before saving Gradle User Home state
2024-04-30T00:47:28.4164741Z Forcing cache cleanup.
2024-04-30T00:47:29.4964870Z [command]/usr/bin/gradle -g /home/runner/.gradle --no-daemon --build-cache --no-scan --quiet -DGITHUB_DEPENDENCY_GRAPH_ENABLED=false noop
2024-04-30T00:47:45.1255310Z ##[group]Caching Gradle state
2024-04-30T00:47:45.1262624Z Cache hit occurred on the cache key gradle-home-v1|Linux|build[19d2a25c058a8c4bf9a3f41868f2ff57]-ea684c64e7e00b271af718376e983ce7d88814f3, not saving cache.
2024-04-30T00:47:45.1264542Z ##[endgroup]
2024-04-30T00:47:45.1270460Z Generating Job Summary
2024-04-30T00:47:45.1287359Z Completed post-action step
2024-04-30T00:47:45.1493826Z Post job cleanup.
2024-04-30T00:47:45.3460542Z Post job cleanup.
2024-04-30T00:47:45.4558774Z [command]/usr/bin/git version
2024-04-30T00:47:45.4619745Z git version 2.43.2
2024-04-30T00:47:45.4671474Z Temporarily overriding HOME='/home/runner/work/_temp/4fe9cfc3-bfad-462a-95fb-3ec37d3c749e' before making global git config changes
2024-04-30T00:47:45.4673695Z Adding repository directory to the temporary git global config as a safe directory
2024-04-30T00:47:45.4677185Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/Anki-Android-Backend/Anki-Android-Backend
2024-04-30T00:47:45.4714233Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2024-04-30T00:47:45.4747050Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2024-04-30T00:47:45.4996205Z Entering 'anki'
2024-04-30T00:47:45.5053005Z Entering 'anki/ftl/core-repo'
2024-04-30T00:47:45.5171501Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2024-04-30T00:47:45.5196833Z http.https://github.com/.extraheader
2024-04-30T00:47:45.5218962Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2024-04-30T00:47:45.5262173Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2024-04-30T00:47:45.5562037Z Entering 'anki'
2024-04-30T00:47:45.5626498Z Entering 'anki/ftl/core-repo'
2024-04-30T00:47:45.6246453Z Cleaning up orphan processes
2024-04-30T00:47:45.6779494Z Terminate orphan process: pid (7901) (java)
```

**Log diff (rough)**

```diff
# ##[group]Run cargo run -p build_rust
# �[36;1mcargo run -p build_rust�[0m
# shell: /usr/bin/bash -e {0}
# env:
#   DEBUG: 1
#   ANDROID_NDK_HOME: /usr/local/lib/android/sdk/ndk/26.3.11579264
#   ANDROID_NDK_ROOT: /usr/local/lib/android/sdk/ndk/26.3.11579264
#   JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/21.0.2-13/x64
#   JAVA_HOME_21_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/21.0.2-13/x64
#   GRADLE_ACTION_ID: gradle/actions/setup-gradle
#   GRADLE_BUILD_ACTION_SETUP_COMPLETED: true
#   GRADLE_BUILD_ACTION_CACHE_RESTORED: true
#   DEVELOCITY_INJECTION_INIT_SCRIPT_NAME: gradle-actions.inject-develocity.init.gradle
#   DEVELOCITY_AUTO_INJECTION_CUSTOM_VALUE: gradle-actions
#   GITHUB_DEPENDENCY_GRAPH_ENABLED: false
# ##[endgroup]
#    Compiling anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
#    Compiling anki_process v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/process)
#    Compiling build_rust v0.1.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/build_rust)
#     Finished dev [unoptimized + debuginfo] target(s) in 0.50s
#      Running `target/debug/build_rust`
# *** Building desktop web components
# �[1m�[32m   Compiling�[0m runner v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/build/runner)
# �[1m�[32m   Compiling�[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
# �[1m�[32m   Compiling�[0m anki_process v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/process)
# �[1m�[32m    Finished�[0m release [optimized] target(s) in 3.64s
# �[1m�[32m   Compiling�[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
# �[1m�[32m   Compiling�[0m ninja_gen v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/build/ninja_gen)
# �[1m�[32m   Compiling�[0m configure v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/build/configure)
# �[1m�[32m    Finished�[0m dev [unoptimized + debuginfo] target(s) in 0.95s
# �[1m�[32m     Running�[0m `out/rust/debug/configure`
# build:configure_bin
# �[1m�[32m    Finished�[0m dev [unoptimized + debuginfo] target(s) in 0.13s
+ build:configure
+ �[1m�[32m    Finished�[0m dev [unoptimized + debuginfo] target(s) in 0.11s
+ �[1m�[32m     Running�[0m `out/rust/debug/configure`
# build:configure
# download:protoc
# download:node
# ftl:repo:core
# extract:protoc
# extract:node
# rslib:proto
# yarn
# node_modules
# rslib:proto
# �[1m�[32m   Compiling�[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
# �[1m�[32m   Compiling�[0m anki_proto_gen v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/proto_gen)
# �[1m�[32m   Compiling�[0m anki_proto v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/proto)
# �[1m�[32m    Finished�[0m dev [unoptimized + debuginfo] target(s) in 2.18s
# ftl:repo:core
# Submodule 'ftl/core-repo' (https://github.com/ankitects/anki-core-i18n.git) registered for path 'ftl/core-repo'
# Cloning into '/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/ftl/core-repo'...
# From https://github.com/ankitects/anki-core-i18n
#  * branch              e3af3c983241448a239871ca573c9dd2fa5e8619 -> FETCH_HEAD
# Submodule path 'ftl/core-repo': checked out 'e3af3c983241448a239871ca573c9dd2fa5e8619'
# ftl:repo:qt
+ �[0m�[1m�[31m
+ Build failed.
+ Error: Failed with code Some(1): ./ninja extract:protoc css:_root-vars ts:reviewer:reviewer.js ts:reviewer:reviewer.css ts:reviewer:reviewer_extras_bundle.js ts:reviewer:reviewer_extras.css qt:aqt:data:web:pages
+ �[0m
+ ##[error]Process completed with exit code 1.
+ Post job cleanup.
+ In post-action step
+ Stopping all Gradle daemons before saving Gradle User Home state
+ Forcing cache cleanup.
+ [command]/usr/bin/gradle -g /home/runner/.gradle --no-daemon --build-cache --no-scan --quiet -DGITHUB_DEPENDENCY_GRAPH_ENABLED=false noop
+ ##[group]Caching Gradle state
+ Cache hit occurred on the cache key gradle-home-v1|Linux|build[19d2a25c058a8c4bf9a3f41868f2ff57]-ea684c64e7e00b271af718376e983ce7d88814f3, not saving cache.
+ ##[endgroup]
+ Generating Job Summary
+ Completed post-action step
+ Post job cleanup.
+ Post job cleanup.
+ [command]/usr/bin/git version
+ git version 2.43.2
+ Temporarily overriding HOME='/home/runner/work/_temp/4fe9cfc3-bfad-462a-95fb-3ec37d3c749e' before making global git config changes
+ Adding repository directory to the temporary git global config as a safe directory
+ [command]/usr/bin/git config --global --add safe.directory /home/runner/work/Anki-Android-Backend/Anki-Android-Backend
+ [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
+ [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
+ Entering 'anki'
+ Entering 'anki/ftl/core-repo'
+ [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
+ http.https://github.com/.extraheader
+ [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
+ [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
+ Entering 'anki'
+ Entering 'anki/ftl/core-repo'
+ Cleaning up orphan processes
+ Terminate orphan process: pid (7901) (java)
- node_modules
- �[2K�[1G�[1myarn install v1.22.22�[22m
- �[2K�[1G�[2m[1/4]�[22m Resolving packages...
- �[2K�[1G�[32msuccess�[39m Already up-to-date.
- �[2K�[1GDone in 0.36s.
- ts:lib:proto
- ts:lib:proto
- css:_root-vars
- ts:reviewer:reviewer_extras.css
- ts:reviewer:reviewer.css
- ftl:repo:qt
- Submodule 'ftl/qt-repo' (https://github.com/ankitects/anki-desktop-ftl.git) registered for path 'ftl/qt-repo'
- Cloning into '/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/ftl/qt-repo'...
- From https://github.com/ankitects/anki-desktop-ftl
-  * branch            45155310c3302cbbbe645dec52ca196894422463 -> FETCH_HEAD
- Submodule path 'ftl/qt-repo': checked out '45155310c3302cbbbe645dec52ca196894422463'
- rslib:i18n
- �[1m�[32m   Compiling�[0m anki_io v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/io)
- �[1m�[32m   Compiling�[0m anki_i18n v0.0.0 (/home/runner/work/Anki-Android-Backend/Anki-Android-Backend/anki/rslib/i18n)
- �[1m�[32m    Finished�[0m dev [unoptimized + debuginfo] target(s) in 1.74s
- ts:import-csv
- ts:image-occlusion
- ts:graphs
- ts:import-page
- ts:congrats
- ts:editable
- ts:reviewer:reviewer_extras_bundle.js
- ts:change-notetype
- ts:card-info
- ts:deck-options
- ts:reviewer:reviewer.js
-ts:import-anki-package
- qt:aqt:data:web:pages
- n2: ran 30 tasks, now up to date
- �[0m�[1m�[32m
- Build succeeded in 23.98s.
- �[0m*** Building Android JNI library + backend interface
```

--- a/.github/workflows/build-quick.yml
+++ b/.github/workflows/build-quick.yml
@@ -112,7 +112,15 @@ jobs:
           gradle-home-cache-cleanup: true
 
       - name: Build all (current platform)
-        run: cargo run -p build_rust
+        # Ubuntu currently requires multiple build attempts before it succeeds - it builds a little further each time then works
+        # This happens the same way every time in CI so far, and is reproducible locally if you have an Ubuntu system
+        # This needs a root-cause analysis (see #373) but we will use a retry hack for Ubuntu only to unblock PRs until that is done
+        uses: nick-fields/retry@v3
+        with:
+          timeout_minutes: 20
+          max_attempts: ${{ contains(matrix.os, 'ubuntu') == true && 5 || 1 }}
+          retry_on: error
+          command: cargo run -p build_rust
 
       - name: Check Rust (Unix)
         if: contains(matrix.os, 'windows') == false

